machine/v1beta1/types_azureprovider.go

```diff
@@ -183,6 +183,20 @@ type OSDisk struct {
 	ManagedDisk ManagedDiskParameters `json:"managedDisk"`
 	// DiskSizeGB is the size in GB to assign to the data disk.
 	DiskSizeGB int32 `json:"diskSizeGB"`
+	// EphemeralDiskSettings describe ephemeral disk settings for the os disk.
+	// +optional
+	EphemeralDiskSettings EphemeralDiskSettings `json:"ephemeralDiskSettings,omitempty"`
```

Comment:
Does calling this `EphemeralDiskSettings` limit us in the future? Are there potentially any options we would want to add to this struct that aren't to do with ephemeral disks? Should this just be `DiskSettings` and then the `Ephemeral` option can be within it?